blockchain/chain_sync/src/sync.rs

```diff
@@ -947,7 +992,7 @@ mod tests {
             Cid::from_raw_cid("bafy2bzacecujyfvb74s7xxnlajidxpgcpk6abyatk62dlhgq6gcob3iixhgom")
                 .unwrap();
 
-        let root = cs.compute_msg_data(&[bls], &[secp]).unwrap();
+        let root = ChainSyncer::<MemoryDB>::compute_msg_data(&[bls], &[secp]).unwrap();
```

Comment:
Reason for it being associated?

Reply:
For the `compute_msg_data` function, `self` had to be removed as one of the parameters in order to get `spawn_blocking` to work. As a result of this, `root` was refactored in that way.